### PR TITLE
fixes https://rollbar.com/xxxmandoxxx/entity-office/items/352

### PR DESCRIPTION
**Pull request by bapu-nexl**
Defined a set of fields we care about. We are getting rollbars for any new field added by SNOV because we don't have accessors. By defining this set of supported fields we can avoid the rollbars.

Looks a little ugly.

### CODE REVIEW

**Comment by grantspeelman** [comment]
Maybe we switch to `CamelSnakeStruct`, I started it here : https://github.com/NEXL-LTS/snov-ruby/blob/main/lib/snov/domain_search.rb#L6
Was on my wish list to switch the others as well.
What do you think?

**Comment by bapu-nexl** [comment]
> Maybe we switch to `CamelSnakeStruct`, I started it here : https://github.com/NEXL-LTS/snov-ruby/blob/main/lib/snov/domain_search.rb#L6
> Was on my wish list to switch the others as well.
> What do you think?

Right  that will work

File: lib/snov/get_emails_by_social_url.rb [hunk 1]
```diff
@@ -18,13 +18,7 @@ def raw_result
...
     end
 
-    class ProspectJob
-      include ActiveModel::Model
-
-      attr_accessor :company_name, :company_type, :position, :social_link, :site, :locality, :state,
-                    :city, :street, :street2, :country, :start_date, :end_date, :postal, :founded, :size,
-                    :industry, :hq_phone
-    end
+    ProspectJob = CamelSnakeStruct
```
Comment (grantspeelman) [inline lib/snov/get_emails_by_social_url.rb:21]:
```suggestion
    ProspectJob = Class.new(CamelSnakeStruct)
```

File: lib/snov/get_profile_by_email.rb [hunk 2]
```diff
@@ -37,13 +37,7 @@ def raw_result
...
     end
 
-    class Job
-      include ActiveModel::Model
-
-      attr_accessor :company_name, :position, :social_link, :site, :locality, :state, :city
-      attr_accessor :street, :street2, :postal, :founded, :start_date, :end_date, :size
-      attr_accessor :industry, :company_type, :country, :hq_phone
-    end
+    Job = CamelSnakeStruct
```
Comment (grantspeelman) [inline lib/snov/get_profile_by_email.rb:40]:
```suggestion
    Job = Class.new(CamelSnakeStruct)
```

File: lib/snov/get_prospects_by_email.rb [hunk 3]
```diff
@@ -30,13 +30,7 @@ def raw_result
...
     end
 
-    class Job
-      include ActiveModel::Model
-
-      attr_accessor :company_name, :position, :social_link, :site, :locality, :state, :city
-      attr_accessor :street, :street2, :postal, :founded, :start_date, :end_date, :size
-      attr_accessor :industry, :company_type, :country, :hq_phone
-    end
+    Job = CamelSnakeStruct
```
Comment (grantspeelman) [inline lib/snov/get_prospects_by_email.rb:33]:
```suggestion
    Job = Class.new(CamelSnakeStruct)
```

**Review by grantspeelman — COMMENTED** [review]
Best to use sub classes instead on Class Aliases so if there are any bugs it's clear which Class is doing it

**Review by grantspeelman — APPROVED** [review]
Nice One 👍🏽

File: lib/snov.rb [hunk 4]
```diff
@@ -33,6 +33,7 @@ def self.use_fake?
 require 'snov/in_memory_token_storage'
 require 'snov/redis_token_storage'
 require 'snov/fake_client'
+require 'camel_snake_struct'
```
Comment (kai-is-coding) [inline lib/snov.rb:36]: Question: what is the purpose of using `camel_snake_struct`? what does this repo `snov-ruby` do?

**Pull request merged**